### PR TITLE
Add sms_region_config block to identity platform.

### PR DESCRIPTION
**Pull request by ascherkus**
Permits configuring the allow/denylist for SMS regions.

Addresses a portion of https://github.com/hashicorp/terraform-provider-google/issues/14194

```release-note:enhancement
identityplatform: added `sms_region_config` to the resource `google_identity_platform_config`
```

### CODE REVIEW

**Comment by ascherkus** [comment]
FYI I tried to set the `conflicts` fields but I couldn't quite figure out why the generated go code didn't update. Pointers appreciated!

File: mmv1/third_party/terraform/services/identityplatform/resource_identity_platform_config_test.go [hunk 1]
```diff
@@ -82,6 +82,13 @@ resource "google_identity_platform_config" "basic" {
         }
     }
   }
+  sms_region_config {
+    allow_by_default {
+      disallowed_regions = [
+        "US",
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/identityplatform/resource_identity_platform_config_test.go:88]: As the `disallowed_regions` is a list, can you please add more items to the list?

File: mmv1/third_party/terraform/services/identityplatform/resource_identity_platform_config_test.go [hunk 2]
```diff
@@ -122,6 +129,13 @@ resource "google_identity_platform_config" "basic" {
         }
     }
   }
+  sms_region_config {
+    allowlist_only {
+      allowed_regions = [
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/identityplatform/resource_identity_platform_config_test.go:134]: As the `allowed_regions` is a list, can you please add more items to the list?

File: mmv1/third_party/terraform/services/identityplatform/resource_identity_platform_config_test.go [hunk 3]
```diff
@@ -82,6 +82,13 @@ resource "google_identity_platform_config" "basic" {
         }
     }
   }
+  sms_region_config {
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/identityplatform/resource_identity_platform_config_test.go:85]: Can you please add `sms_region_config` with `allow_by_default` or `allowlist_only` to the example? https://github.com/GoogleCloudPlatform/magic-modules/blob/4b727fa6623e03eef4774ddfaae84b9b0cc74202/mmv1/templates/terraform/examples/identity_platform_config_basic.tf.erb

**Comment by zli82016** [comment]
> FYI I tried to set the `conflicts` fields but I couldn't quite figure out why the generated go code didn't update. Pointers appreciated!

added a comment in the code

File: mmv1/third_party/terraform/services/identityplatform/resource_identity_platform_config_test.go (hunk 3 above)
Comment (ascherkus) [reply mmv1/third_party/terraform/services/identityplatform/resource_identity_platform_config_test.go:85]: Done!

File: mmv1/third_party/terraform/services/identityplatform/resource_identity_platform_config_test.go (hunk 2 above)
Comment (ascherkus) [reply mmv1/third_party/terraform/services/identityplatform/resource_identity_platform_config_test.go:135]: Done!

File: mmv1/third_party/terraform/services/identityplatform/resource_identity_platform_config_test.go (hunk 1 above)
Comment (ascherkus) [reply mmv1/third_party/terraform/services/identityplatform/resource_identity_platform_config_test.go:89]: Done!

File: mmv1/products/identityplatform/Config.yaml [hunk 4]
```diff
@@ -223,3 +223,32 @@ properties:
...
     default_from_api: true
+  - !ruby/object:Api::Type::NestedObject
+    name: 'smsRegionConfig'
+    description: |
+      Configures the regions where users are allowed to send verification SMS for the project or tenant. This is based on the calling code of the destination phone number.
+    properties:
+      - !ruby/object:Api::Type::NestedObject
+        name: 'allowByDefault'
+        conflicts:
+          - smsRegionConfig.allowlistOnly
```
[the remark this answers is not in the document]
Comment (ascherkus) [reply mmv1/products/identityplatform/Config.yaml]: I changed it to:
```yaml
        exactly_one_of:
          - sms_region_config.0.allow_by_default
          - sms_region_config.0.allowlist_only
```

... and tested that it has the correct behaviour (I temporarily modified the generated and test and verified all possible combinations)

File: mmv1/products/identityplatform/Config.yaml [hunk 5]
```diff
@@ -223,3 +223,32 @@ properties:
...
     item_type: Api::Type::String
     default_from_api: true
+  - !ruby/object:Api::Type::NestedObject
+    name: 'smsRegionConfig'
+    description: |
+      Configures the regions where users are allowed to send verification SMS for the project or tenant. This is based on the calling code of the destination phone number.
+    properties:
+      - !ruby/object:Api::Type::NestedObject
+        name: 'allowByDefault'
+        conflicts:
```
[the remark this answers is not in the document]
Comment (ascherkus) [reply mmv1/products/identityplatform/Config.yaml]: Done

**Comment by ascherkus** [comment]
Thanks for the review. Addressed all feedback and tested the new `exactly_one_of` behaviour.

**Review by zli82016 — APPROVED** [review]
LGTM

**Pull request merged**